src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -81,9 +83,74 @@ var {
   resetContext,
 } = require('ReactFiberContext');
 
+var onCommitRootInDev = null;
+var callCommitListenerInDev = null;
+
 if (__DEV__) {
   var ReactFiberInstrumentation = require('ReactFiberInstrumentation');
   var ReactDebugCurrentFiber = require('ReactDebugCurrentFiber');
+  var warning = require('warning');
+
+  var devCommitListeners = [];
```

Comment:
In practice, since this only attaches to `__REACT_DEVTOOLS_GLOBAL_HOOK__` there will only ever need to be one listener, which is the `__REACT_DEVTOOLS_GLOBAL_HOOK__` itself. It might be simpler just do track it as `const listener = __REACT_DEVTOOLS_GLOBAL_HOOK__` and call on it.

That way this source can be smaller for when we put it in prod.